src/main/java/uk/gov/pay/ledger/transaction/service/TransactionMetadataService.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.pay.ledger.transaction.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.ledger.event.model.Event;
+import uk.gov.pay.ledger.metadatakey.dao.MetadataKeyDao;
+import uk.gov.pay.ledger.transaction.dao.TransactionDao;
+import uk.gov.pay.ledger.transactionmetadata.dao.TransactionMetadataDao;
+
+import java.util.Iterator;
+
+public class TransactionMetadataService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransactionMetadataService.class);
+    private final TransactionDao transactionDao;
+    private final MetadataKeyDao metadataKeyDao;
+    private final TransactionMetadataDao transactionMetadataDao;
+
+    @Inject
+    public TransactionMetadataService(MetadataKeyDao metadataKeyDao,
+                                      TransactionMetadataDao transactionMetadataDao,
+                                      TransactionDao transactionDao) {
+        this.metadataKeyDao = metadataKeyDao;
+        this.transactionMetadataDao = transactionMetadataDao;
+        this.transactionDao = transactionDao;
+    }
+
+    public void upsertMetadataFor(Event event) {
+        JsonNode eventDataNode;
+        try {
+            eventDataNode = new ObjectMapper().readTree(event.getEventData());
+        } catch (JsonProcessingException e) {
+            LOGGER.error("Unable to parse incoming event payload: {}", e.getMessage());
+            return;
```

Comment:
Can we throw exception if unable to parse event payload ? If not, there is a chance we won't have visibility for errors.

if exception is thrown, events will be placed in dead letter queue for which we will be alerted

Reply:
Given that exception is thrown https://github.com/alphagov/pay-ledger/blob/66b7312a20e0046d3d33c9e9f6c5c5dda9256f7c/src/main/java/uk/gov/pay/ledger/event/model/EventDigest.java#L85 for invalid JSON we might not need to throw exceptions in later stages.